### PR TITLE
Add Graphlient::Errors::ConnectionFailedError

### PR DESCRIPTION
**Pull request by neroleung**
This is essentially https://github.com/ashkan18/graphlient/pull/59 with test and some other dependency updates.

Whenever there is an `Errno::ECONNREFUSED` error, Faraday wraps it in a `Faraday::ConnectionFailed` error. Since it inherits `Faraday::ClientError` without a response (see errors/faraday_server_error.rb), the code would return a `NoMethodError: undefined method `[]' for nil:NilClass` error.

So basically the fix is to `rescue Faraday::ConnectionFailed` before rescuing `Faraday::ClientError` so that we can raise a more meaningful error.

Result:
```
Graphlient::Errors::ConnectionFailedError: Failed to open TCP connection to localhost:3000 (Connection refused - connect(2) for "localhost" port 3000)
```

Some other updates:
- Updating ruby to 2.6.5 because zeitwerk requires Ruby version >= 2.4.4.
- I added pry as a dev dependency because I think it's more useful than byebug.
  I can take it out if anyone disagree.
- Minor rubocop fixes.

### CODE REVIEW

**Comment by neroleung** [comment]
It doesn't seem like I can add reviewers here. Not sure why. But @ashkan18 and @yuki24, please take a look this what you have a moment. I'd hate for my company to drop your gem because of this simple issue. 🙇

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.3.0
\ No newline at end of file
+2.6.5
```
Comment (yuki24) [inline .ruby-version]: I believe dropping a Ruby version should be a separate PR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
-### 0.3.7 (Next)
+### 0.3.8 (Next)
 * Your contribution here.
 
+### 0.3.7 (Next)
```
Comment (yuki24) [inline CHANGELOG.md]: Individual pull requests should not update version numbers as each new version could have multiple changes. Depending on the activity, a release may end up with just one change, but it is not something we decide on as part of a review process.

File: lib/graphlient/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Graphlient
-  VERSION = '0.3.6'.freeze
+  VERSION = '0.3.7'.freeze
```
Comment (yuki24) [inline lib/graphlient/version.rb]: Ditto https://github.com/ashkan18/graphlient/pull/66/files#r343998878

File: .ruby-version (hunk 1 above)
Comment (neroleung) [reply .ruby-version]: In internal development, I'd agree. But for public PRs, when what we really wanna do is to fix things up and move on, I'd love to see flexibility there. That being said, if it's a blocker, I can submit separate PRs.

File: CHANGELOG.md (hunk 2 above)
Comment (neroleung) [reply CHANGELOG.md]: I apologize. I was trying to follow https://github.com/ashkan18/graphlient/pull/59. I will take it out.

File: Gemfile [hunk 4]
```diff
@@ -5,9 +5,8 @@ gemspec
 gem 'rake'
 
 group :development do
-  gem 'byebug', platform: :ruby
   gem 'danger-changelog', '~> 0.2.1'
-  gem 'rubocop', '0.56.0'
+  gem 'rubocop'
```
Comment (yuki24) [inline Gemfile]: Rubocop upgrades should be a separate pull request as it often involves more discussions around preferable syntax/pattern in the project.

File: Gemfile [hunk 5]
```diff
@@ -21,3 +20,8 @@ group :test do
   gem 'vcr'
   gem 'webmock'
 end
+
+group :development, :test do
+  gem 'byebug'
+  gem 'pry'
+end
```
Comment (yuki24) [inline Gemfile]: Same here, one thing at a time. Adding gems as part of a pull request for something else is generally not a good idea.

**Comment by yuki24** [comment]
Thanks for your PR! I like the idea of `Graphlient::Errors::ConnectionFailedError`. @ashkan18 @dblock WDYT?

Also, as a general rule, one pull request should not include any other changes then the proposed solution. Extra changes in one PR would make it harder to revert and tends to make the conversation longer, so "less is more" also applies here.

> Updating ruby to 2.6.5 because zeitwerk requires Ruby version >= 2.4.4.

I don't think we should do this as part of this change. `graphlient` does not depend on Rails and zeitwerk is outside of the gem's concern.

> I added pry as a dev dependency because I think it's more useful than byebug.

`pry` and `byebug` are two distinct tools that have different behavior (`pry` is compatible with JRuby whereas `byebug` is MRI-only but has better thread support) and I would pick one depending on the situation.

> It doesn't seem like I can add reviewers here

No worries, we all get notified when a new pull request or issue is posted.

**Comment by dblock** [comment]
This is actually a useful change, thanks for bringing this up. Care to clean it up per @yuki24's comments and reopen @neroleung?

File: .ruby-version (hunk 1 above)
Comment (yuki24) [reply .ruby-version]: In open source, it is very important to respect the maintainer's preferences otherwise we would end up hearing conflicting opinions from different people who want to push their preferences. In this project, @ashkan18 and @dblock are the decision makers (I'm just helping them run the project) and I'll leave all preference decisions up to them.

In this specific instance, it is understandable to use the least supported version so any new API introduced to Ruby won't get creeped in. I'm not also sure how it would make it more flexible since the version constraints are still there, and people who care about compatibility and want to test against 2.3 will have to change it back. I don't understand why and how the visibility of the project would affect the versioning policy.

My preference is to not have a `.ruby-version` at all so I can more easily test the library with different Ruby versions locally. However, just changing it up to the latest version sounds like one's preference is being pushed up.
